pkg/cloudprovider/azure.go

```diff
@@ -586,13 +586,14 @@ func (a *Azure) getAuthorizer(env azureapi.Environment, cfg *azureCredentialsCon
 		err  error
 	)
 
-	// MSI Override for ARO HCP
-	msi := os.Getenv("AZURE_MSI_AUTHENTICATION")
-	if msi == "true" {
+	// Managed Identity Override for ARO HCP
+	managedIdentityClientID := os.Getenv("ARO_HCP_MI_CLIENT_ID")
+	if managedIdentityClientID != "" {
 		options := azidentity.ManagedIdentityCredentialOptions{
 			ClientOptions: azcore.ClientOptions{
 				Cloud: cloudConfig,
 			},
+			ID: azidentity.ClientID(managedIdentityClientID),
```

Comment:
The mock-http demo seems to work without specifying the ID. Can this field be omitted?

Reply:
Someone I talked to from Microsoft said we would need a valid Client ID here when we are using the adapter sidecar containers. More info on that here - https://github.com/openshift/enhancements/pull/1659#issuecomment-2350008814.